app/javascript/packs/state_file.js

```diff
@@ -1,22 +1,62 @@
 import { createConsumer } from "@rails/actioncable"
 
-const { controllerAction } = document.querySelector("#mixpanelData")?.dataset || {};
-if (controllerAction == "StateFile::Questions::WaitingToLoadDataController#edit") {
-    let channelName = {
-        channel: 'DfDataTransferJobChannel',
-    };
-    createConsumer().subscriptions.create(channelName, {
-        connected() {
-            console.log("We're connected!")
-            document.querySelector('[data-after-data-transfer-button]').setAttribute('data-subscribed', 'true')
-        },
-
-        disconnected() {
-            console.log("We're disconnected!")
-        },
-
-        received(data) {
-            document.querySelector('[data-after-data-transfer-button]').click()
+const actionMap = {
+  "StateFile::Questions::WaitingToLoadDataController#edit": {
+    channelName: { channel: "DfDataTransferJobChannel" },
+    channelEvents: {
+      connected() {
+        console.log("We're connected!")
+        document.querySelector('[data-after-data-transfer-button]').setAttribute('data-subscribed', 'true')
+      },
+
+      disconnected() {
+        console.log("We're disconnected!")
+      },
+
+      received(data) {
+        document.querySelector('[data-after-data-transfer-button]').click()
+      }
+    }
+  },
+  "StateFile::Questions::SubmissionConfirmationController#edit": {
+    channelName: { channel: "StateFileSubmissionPdfStatusChannel" },
+    channelEvents: {
+      initialized () {
+        this.pdfStatus = null
```

Comment:
Do you think it would be reasonable to put a timer here that shows the button (and disconnects the channel) after a few (5+) seconds, regardless of what the channel has reported? Most just as a safety measure in case ActionCable fails to connect or the job doesn't get picked up? At worst, clicking the button when the job hasn't completed will result in falling back to the synchronous PDF generation in `StateFile::Questions::SubmissionPdfsController#show`

Reply:
that a great idea.